tools/mo/openvino/tools/mo/moc_frontend/pytorch_frontend_utils.py

```diff
@@ -0,0 +1,214 @@
+        scripted = model
+    f_model = torch.jit.freeze(scripted)
+    decoder = TorchScriptPythonDecoder(f_model)
+    return decoder, input_signature
+
+
+def to_torch_tensor(tensor):
+    import torch
+    if isinstance(tensor, torch.Tensor):
+        return tensor
+    if isinstance(tensor, np.ndarray):
+        return torch.tensor(tensor)
+    if isinstance(tensor, np.ndarray):
+        return torch.tensor(tensor)
+    if isinstance(tensor, Tensor):
+        return torch.tensor(tensor.data)
+    else:
+        raise Error("Unexpected type of example_input. Supported types torch.Tensor, np.array or ov.Tensor. "
+                    "Got {}".format(type(tensor)))
+
+
+def prepare_torch_inputs(example_inputs, input_shape, allow_none=False):
+    import torch
+    inputs = None
+    if example_inputs is not None:
+        inputs = example_inputs
+        if isinstance(inputs, list):
+            inputs = [to_torch_tensor(x) for x in inputs]
+            if len(inputs) == 1:
+                inputs = torch.unsqueeze(inputs[0], 0)
+            else:
+                inputs = inputs
+        elif isinstance(inputs, tuple):
+            inputs = [to_torch_tensor(x) for x in inputs]
+            inputs = tuple(inputs)
+        elif isinstance(inputs, dict):
+            for name, tensor in inputs.items():
+                assert isinstance(name, str), "Expected dictionary where keys are input names of string type and" \
+                                              " values are tensors. Got key of type {}".format(type(name))
+                inputs[name] = to_torch_tensor(tensor)
+        else:
+            inputs = to_torch_tensor(inputs)
+    elif input_shape is not None:
+        inputs = []
+        for shape in input_shape:
+            static_shape = get_static_shape(shape, dynamic_value=1)
+            inputs.append(torch.zeros(static_shape))
+        inputs = tuple(inputs)
+    else:
+        if not allow_none:
+            raise Error("Please provide input_shape or example_input for converting PyTorch model.")
+    return inputs
+
+
+def convert_pytorch_to_onnx(model, input_shape, opset_version, example_inputs, output_dir):
+    import io
+    import torch
+
+    input_names = None
+    inputs = prepare_torch_inputs(example_inputs, input_shape)
+
+    dynamic_dims_dict = {}
+    if input_shape is not None and input_names is None:
+        input_names = ["input_{}".format(idx) for idx in range(len(input_shape))]
+        for shape_idx, shape in enumerate(input_shape):
+            dynamic_dims = get_dynamic_dims(shape)
+            if len(dynamic_dims) > 0:
+                dynamic_dims_dict[input_names[shape_idx]] = dynamic_dims
+    additional_params = {}
+    if len(dynamic_dims_dict) > 0:
+        additional_params.update({'dynamic_axes': dynamic_dims_dict})
+    if input_names is not None and len(input_names) > 0:
+        additional_params.update({'input_names': input_names})
+
+    if os.environ.get('SAVE_TO_BYTES_IO_ONNX_MODEL'):
```

Comment:
@popovaan, please explain why we need `SAVE_TO_BYTES_IO_ONNX_MODEL`.

Reply:
This variable switches two possible ways to pass onnx model converted from pytorch to inner convert_model(). 
1. Save to temporary onnx file - default way
2. Pass as BytesIO as object in memory - if  `SAVE_TO_BYTES_IO_ONNX_MODEL` is set. Currently only legacy MO supports input_model as BytesIO, so for inner convert_model() `use_legacy_frontend` is set.

Measurements showed that there were no significant advantage in any approach, so both were left as possible conversion way.